### PR TITLE
Profiles

### PR DESCRIPTION
**Pull request by vlevigneron**
Works with corresponding zonemaster::Engine profiles branch.

### CODE REVIEW

**Comment by matsduf** [comment]
@vlevigneron, I would have expected updated CLI to fail against current Engine. Do we lack some unit tests?

**Comment by vlevigneron** [comment]
le 09 nov., Mats Dufberg a ?crit :
> @vlevigneron, I would have expected updated CLI to fail against current Engine. Do we lack some unit tests?
> 

Perhaps, I have not tested it against current version of Engine, only
updated one.

> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/zonemaster/zonemaster-cli/pull/93#issuecomment-437299725

-- 
	Vincent Levigneron  A.F.N.I.C.  Vincent.Levigneron@afnic.fr

File: lib/Zonemaster/CLI.pm [hunk 1]
```diff
@@ -292,37 +307,47 @@ sub run {
...
+        Zonemaster::Engine->profile->load( $self->profile );
     }
+    else {
 
-    if ( $self->config ) {
-        say $fh_diag __x( "Loading configuaration from {path}.", path => $self->config );
-        Zonemaster::Engine->config->load_config_file( $self->config );
+        if ( $self->policy ) {
+            say $fh_diag __x( "Loading policy from {path}.", path => $self->policy );
+            say $fh_diag __x( "DEPRECATED, use profile instead." );
```
Comment (mattias-p) [inline lib/Zonemaster/CLI.pm]: We should use a stronger word than deprecated if we make it an error to use it. Deprecation implies that the old way still works for the time being but will be removed in a future version.
https://en.wikipedia.org/wiki/Deprecation

Comment (matsduf) [reply lib/Zonemaster/CLI.pm]: I suggest that we write "Use of config and policy have been TERMINATED. Use profile instead."

**Comment by matsduf** [comment]
@vlevigneron, Travis has done the tests against the current version. I think it means that it lacks some unit tests when it passes.

File: lib/Zonemaster/CLI.pm (hunk 1 above)
Comment (vlevigneron) [reply lib/Zonemaster/CLI.pm]: Done this way.

**Comment by vlevigneron** [comment]
@matsduf Theer are very few UT in the CLI, compared to the Engine. I agree we should improve this in the future.

**Comment by sandoche2k** [comment]
@matsduf and @mattias-p After the discussion last week.. do you agree to accept this PR waiting for updating the travis test requirements in the future?

**Comment by sandoche2k** [comment]
@mattias-p thanks for reviewing, so that @vlevigneron can merge

**Pull request merged**